mobile/apps/photos/lib/services/video_preview_service.dart

```diff
@@ -89,11 +110,83 @@ class VideoPreviewService {
   void clearQueue() {
     fileQueue.clear();
     _items.clear();
-    _hasQueuedFile = false;
+  }
+
+  Future<bool> addToManualQueue(EnteFile file, String queueType) async {
+    if (file.uploadedFileID == null) return false;
+
+    // Check if already in queue
+    final bool alreadyInQueue =
+        await UploadLocksDB.instance.isInStreamQueue(file.uploadedFileID!);
+    if (alreadyInQueue) {
+      return false; // Indicates file was already in queue
+    }
+
+    // Add to persistent database queue
+    await UploadLocksDB.instance
+        .addToStreamQueue(file.uploadedFileID!, queueType);
+
+    _items[file.uploadedFileID!] = PreviewItem(
+      status: PreviewItemStatus.inQueue,
+      file: file,
+      retryCount: 0,
+      collectionID: file.collectionID ?? 0,
+    );
+    fileQueue[file.uploadedFileID!] = file;
+
+    // Start processing if not already processing
+    if (uploadingFileId < 0) {
+      queueFiles(duration: Duration.zero);
+    }
+
+    return true; // Indicates file was successfully added to queue
```

Comment:
We can document this on top that the method returns True if the file is queued?

Reply:
Done!